CONTRIBUTING.md

```diff
@@ -0,0 +1,135 @@
+
+Where appropriate, commits should always contain tests for the code in question.
+
+#### Running tests (CI script)
+
+Submitted PRs are expected to pass continuous integration (CI), which, among other things, runs a test suite on your PR to make sure that your code doesn't have any bugs.
+
+#### Code Review
+
+Your PR will be automatically assigned to the team member(s) specified in the `codeowners` file, who may either review the PR himself/herself or assign it to another team member. More often than not, a code submission will be met with review comments and changes requested. It's nothing personal, but nobody's perfect; we leave each other review comments all the time.
+
+Fixes to review comments should preferably be pushed as additional commits to make it easier for the reviewer to see the changes. As a final step once the reviewer is happy the author should consider squashing these fixes with the relevant commit.
+
+### Project board
+
+GitHub project boards are used by the maintainers of the majority of our repositories to keep track of progress and organise development priorities.
+
+There may be one or more active project boards for a repository. Typically, one main project is used to manage all tasks corresponding to the main development stream (normally the `master` branch), while a separate project would be used to manage each PoC, milestone and/or prototyping development, and each of them will track a dedicated development branch.
+
+New features which involve a large number of changes may be developed in a dedicated feature branch, but would normally be tracked on the same main project board as the main development branch (normally `master` branch), re-basing it with the main branch regularly and fully testing the feature on its own branch before it is fully approved and merged into the main branch.
+
+The main project boards typically contain the following Kanban columns to track the status of each development task:
+
+* **To do**: new issues which need to be reviewed and evaluated to decide their priority, add labels, clarify, etc
+* **In Progress**: the task is assigned to a person and it is in progress
+* **Needs Review**: the task is considered complete by the assigned developer and so has been sent for peer reviewed
+* **Reviewer approved**: the task has been approved by reviewer/s and is considered ready to be merged
+* **Done**: the PR associated with the task was merged (or the task was completed by any other means)
+
+The project board columns would typically include automation to move the issues between columns upon set actions, for example, if a PR was created which indicated in its description that it resolved a particular issue on the project board (using [GitHub keywords](https://help.github.com/articles/closing-issues-using-keywords)) then that issue would automatically be moved to the `Done` column on the board on PR merge.
+
+## Releases and Changelog
```

Comment:
I would still like to see us automate this via `cargo release`, that does not update the changelog, but we could make a changelog PR the trigger for cargo release?

Reply:
Probably should differentiate release of a lib and an application where we provide binaries.

Reply:
Agreed about the lib & binaries point - I'll update to include something to differentiate.
I didn't want to mention anything about `cargo` so as the guide could be generic enough to be added to all our repos, Rust or other